### ISSUE TITLE
请问下能否支持torch.scatter() op 转换到tflite模型

### ISSUE THREAD

**Issue by xsong-AI** [alibaba/TinyNeuralNetwork#120]
我在tflite generated_ops.py 文件里找到了ScatterNdOperator,
不过好像tensorflow里实现方式和pytorch实现方式不一致，我自己试着加代码来着，没有成功。
如果你们能添加这个scatter op的转换就太感谢了！

**Comment by peterjc123** [comment]
@xsong-AI 可以先把你的用法发我们看看不？既然gather能支持，scatter应该也是可以翻译出来的

**Comment by peterjc123** [comment]
这边可以先讲下各种情况下怎么实现
对于t.scatter(dim, idx, src)
1. 最简单是t为全0，src为常量
这种情况下两个op基本是直接对应关系，只不过参数不一样，所以可以准备一组随机输入(src为全1，t为全0）来跑torch那边的结果，然后torch.nonzero获取indices，再在src上索引一下获得值updates，就可以转换成tf这边的输入格式tf.scatter_nd(indices,update,shape)
2. 稍复杂一些的是t为不为0的常量，而src还是常量
其实在1的基础上，拿到indices除外的所有索引，对t做一下gather_nd，然后加起来即可
3. 更复杂一些的是src非常量

5.

**Comment by xsong-AI** [comment]
> 
def softmax_w_top(x, top):
    values, indices = torch.topk(x, k=top, dim=1)
    x_exp = values.exp_()

    x_exp /= torch.sum(x_exp, dim=1, keepdim=True)
    # The types should be the same already
    # some people report an error here so an additional guard is added
    x.zero_().scatter_(1, indices, x_exp.type(x.dtype)) # B * THW * HW

    return x
我这边就是想要将这个op 转到tflite, 大概就是将矩阵行按top_k排序后，算softmax, 然后将更新后的值插回原来的矩阵里面。
所以scatter的idx是会变的。

**Comment by peterjc123** [comment]
@xsong-AI 我想了下，其实gather和scatter里面index的意义是一样的，所以可以通过https://github.com/alibaba/TinyNeuralNetwork/blob/main/tinynn/converter/operators/torch/aten.py#L2143-L2171 这一段来把pytorch的indices变换成tf这边的indices，然后updates值的索引其实就是scatter构建一个arange的数据的索引跑一个fake output出来，然后把对应轴的数据用给定idx给替换了，其实和前面的逻辑差不多，只不过就是要算两次

**Comment by peterjc123** [comment]
@xsong-AI 让你久等了，前几天在休假，现在已经实现了，可以重新拉一下新的代码试试看
